### PR TITLE
Implement all the three questions

### PR DESCRIPTION
**Pull request by uudashr**

### CODE REVIEW

File: parse-promise.js [hunk 1]
```diff
@@ -1 +1,91 @@
...
+                debug(`sending data index: ${index - 1}`);
+                sendSms(line).catch(function (reason) {
+                    debug(reason);
+                    return logToS3(reason);
+                }).catch(function (err) {
+                    debug(err);
+                });
+            }
+
+            index++;
```
Comment (diorahman) [inline parse-promise.js]: Hi @uudashr what is the purpose of this statement?

Comment (uudashr) [inline parse-promise.js]: 1. Transform each the line, merge the "first name" & "last name" into "full name" (except the first line of course). I actually create "parseCsv" as promise style of "parse" function, that block is the "then" block.
2. Send each line to sms, and log if it failed sent.

Comment (diorahman) [inline parse-promise.js]: I meant the `index++`, do you really need that?

Comment (uudashr) [inline parse-promise.js]: Ahh... I see.. Sorry we don't need that. "for in" supposed to be already returned the key/index. My mistakes.

**Pull request closed without merging**